### PR TITLE
docs: add the description of versioning rule

### PR DESCRIPTION
**Pull request by da1suk8**
# Description

Added the description of versioning rule.

## Types of changes

- [ ] Bug fix (changes which fixes an issue)
- [ ] New feature (changes which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] ETC (build, ci, docs, perf, refactor, style, test)

## Checklist

- [x] I followed the [contributing guidelines](https://github.com/Finschia/cosmwasm/blob/main/CONTRIBUTING.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] The PR title and commits are followed [conventional commit form](https://www.conventionalcommits.org/en/v1.0.0).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -39,3 +39,8 @@ TBD
 4. You may merge the pull request in once you have the sign-off of two other developers, or if you
    do not have permission to do that, you may request the second reviewer to merge it for you.
 
+## Releases
+
+Release is maintained in a release branch named `vX.X.X+Y.Y.Y-ZZZ` where `vX.X.X` is the original CosmWasm's release version, `Y.Y.Y` is our release version and `ZZZ` is additional information (e.g. rc, alpha).
```
Comment (loloicci) [inline CONTRIBUTING.md:44]: aren't Y and Z reversed?

Comment (da1suk8) [reply CONTRIBUTING.md]: Updated, thank you!

**Review by zemyblue — APPROVED** [review]

**Review by zemyblue — COMMENTED** [review]
Please add changelog

**Comment by da1suk8** [comment]
@zemyblue 

In cosmwasm, CHANGELOG is generated automatically and does not need to be added.

**Comment by zemyblue** [comment]
> @zemyblue
> 
> In cosmwasm, CHANGELOG is generated automatically and does not need to be added.

If then, what's this? 
https://github.com/Finschia/cosmwasm/blob/main/CHANGELOG.md

These changes are written automatically?

**Comment by da1suk8** [comment]
@zemyblue 

[This](https://github.com/Finschia/cosmwasm/blob/main/devtools/update_changelog.sh) is used to create the CHANGELOG.

However, it does not seem to have been updated since 0.16.3-0.5.1, so a separate PR with an updated CHANGELOG is created.

**Review by zemyblue — APPROVED** [review]

**Review by kokeshiM0chi — APPROVED** [review]

**Pull request merged**